src/libstore/build.cc

```diff
@@ -4415,7 +4425,15 @@ void Worker::waitForInput()
 
 unsigned int Worker::exitStatus()
 {
-    return timedOut ? 101 : (permanentFailure ? 100 : 1);
+    unsigned int mask = 0;
+    if (timedOut)
+        mask |= 1;
+    if (hashMismatch)
+        mask |= 2;
+    if (checkMismatch)
+        mask |= 4;
+
+    return mask ? 100 + mask : 1;
```

Comment:
Hm, using 100 as a base seems wrong in combination with the ORs, since `100 | 4 == 100`.

Maybe it's better to drop adding 100 and just use masks 2, 4, 8, ... since that's unambiguous with the default exit code of 1.

Reply:
The offset is a bit awkward, but changing the existing exit codes would break some things that depend on it. Like some of the ofborg functionality. I didn't really look into it but perhaps the current value can be encoded in a reasonable way?